pymc/distributions/mixture.py

```diff
@@ -438,25 +444,86 @@ def marginal_mixture_logcdf(op, value, rng, weights, *components, **kwargs):
     return mix_logcdf
 
 
-@_get_moment.register(MarginalMixtureRV)
-def get_moment_marginal_mixture(op, rv, rng, weights, *components):
+@_moment.register(MarginalMixtureRV)
+def marginal_mixture_moment(op, rv, rng, weights, *components):
     ndim_supp = components[0].owner.op.ndim_supp
     weights = at.shape_padright(weights, ndim_supp)
     mix_axis = -ndim_supp - 1
 
     if len(components) == 1:
-        moment_components = get_moment(components[0])
+        moment_components = moment(components[0])
 
     else:
         moment_components = at.stack(
-            [get_moment(component) for component in components],
+            [moment(component) for component in components],
             axis=mix_axis,
         )
 
-    moment = at.sum(weights * moment_components, axis=mix_axis)
+    mix_moment = at.sum(weights * moment_components, axis=mix_axis)
     if components[0].dtype in discrete_types:
-        moment = at.round(moment)
-    return moment
+        mix_moment = at.round(mix_moment)
+    return mix_moment
+
+
+# List of transforms that can be used by Mixture, either because they do not require
+# special handling or because we have custom logic to enable them. If new default
+# transforms are implemented, this list and function should be updated
+allowed_default_mixture_transforms = (
+    transforms.CholeskyCovPacked,
+    transforms.CircularTransform,
+    transforms.IntervalTransform,
+    transforms.LogTransform,
+    transforms.LogExpM1,
+    transforms.LogOddsTransform,
+    transforms.Ordered,
+    transforms.Simplex,
+    transforms.SumTo1,
+)
+
+
+@_default_transform.register(MarginalMixtureRV)
+def marginal_mixture_default_transform(op, rv):
+    rng, weights, *components = rv.owner.inputs
+
+    default_transforms = [
+        _default_transform(component.owner.op, component) for component in components
+    ]
+
+    # If there are more than one type of default transforms, we do not apply any
+    if len({type(transform) for transform in default_transforms}) != 1:
```

Comment:
What happens if components don't have the same support? My understanding from this is that we do not perform any variable transformation.

I guess a follow-up question is if it is discouraged to have components with different supports. If so, would it be good to raise a warning?

Reply:
If they don't have the same default transform we don't try to apply any, which is how things worked so far.

Note that distributions may have the same support, but have different default transforms (say logodds vs Interval(0, 1)) and we still won't apply any transform. That's just for simplicity.

Whether people should use distributions with different supports... I think it depends on what the goal is. I don't know if we should raise a warning... Good question

Reply:
It is valid to mix components with different support. Some zero inflated distributions mix a Dirac delta at zero with a distribution that doesn’t include zero in its support. Imagine a zero inflated half normal, if we apply the half normal’s default log transform to 0, we get -inf, which would then cause problems down the line.

Reply:
Maybe we could emit a warning saying that the mixture has components with different support, but it should be a new dedicated warming type, so that it’s easy to catch and silence

Reply:
> It is valid to mix components with different support. Some zero inflated distributions mix a Dirac delta at zero with a distribution that doesn’t include zero in its support. Imagine a zero inflated half normal, if we apply the half normal’s default log transform to 0, we get -inf, which would then cause problems down the line.

That example which mixes discrete and continuous should actually not be allowed until we properly implement such logp (#4511), and they would only work for observed variables not unobserved (#3582) as they would need a specialized sampler. 

Otherwise discrete / continuous mixtures with different support should sample alright. Ideally (for continuous), the user would manually apply a transform that unconstrained the combined support of all the components. We simply won't try such thing here. I think I'll add the warning you mentioned